RFC-008-release-versioning.md

```diff
@@ -0,0 +1,116 @@
+
+1. The PSC and Working Groups will decide on what functionality/features to target for it.
+1. Working Groups will work towards these goals during regular working sessions.
+1. Once the updates to a Product Output are **functionally complete**, meaning the initial implementation of new functionality has been completed, the Product Output can be tagged with a semantic target version number depending on changes introduced.
+1. The first such tag within a Release cycle should also carry the "-alpha" suffix.
+1. Further work during the release will focus on stabilizing the existing functionality and making it ready for general/production use.
+1. Once the Product Output is deemed ready, it will be tagged with the same semantic version number and the "-beta" suffix.
+1. At this point it should be released to a community of beta testers to actually use the Product Output. Feedback from this use should be actively solicited and captured.
+1. Provided there are no "blocker"-level issues issues remaining with the Product Output as the Release's target date approaches, the Release Manager will coordinate the completion of any final updates before signaling for them to be tagged with the same semantic version number, minus any suffix (indicating a final release).
+1. Once tagged, building and packaging the Product Outputs can commence.
+
+{
+This section may continue depending on the answer to the minor version support question below, as in this case we'll be releasing fix versions of the prior release in parallel with the development of the next release.
+}
+
+
+## Consequences
+
+This decision will dictate how Releases are communicated to both internal and external parties. Using the Semantic Versioning convention will be familiar to many of the project's most important audiences, namely open source projects (many of which use the same or similar numbering schemes) and technical writers of various levels of experience.
+
+
+## Dependencies
+
+This RFC should be considered together with RFC-006, which describes the high-level planning process to be used for the project's Releases.
+
+
+## Links and prior art
+
+- [Meeting notes from discussion on versioning](https://docs.google.com/document/d/1Oc3hNhJRrXsr-abEmUKAdamC5aKJydtldvpcSK_Hv_E/edit)
+
+
+## Open questions
+
+- Should we limit Product Outputs to one per Working Group? For example, the Community Docs Working Group would produce the "Template Collection," as opposed to considering each template as a separate Product Output?
+  - In the context of this RFC, this means the collection as a whole would be versioned together.
+  - A result of this would be that, as defined by the SemVer standard, **any** update to a template that adds materially to it (i.e. not fixing typos or other "bugs") would mean an increase in the minor version of the collection as a whole.
+- How long will the project aim to support legacy major versions? Minor versions?
+

```

Comment:
Question (non-blocking): Do we need to worry about what to call releases that are in progress? Like will we use a code name to refer to the release in the community? If so, I propose we use an alphabetic naming scheme (first code name release is A, second code name release is B) and then name each code release after an author starting with that name, such as Austen, Baldwin, Camus, etc. We can pick the next 5 release names at a time through community vote possibly.

I don't want this to be a contentious suggestion so I will gladly withdraw it if it stirs up too much controversy.

Reply:
💯  I've done inventor(s) and my-little ponies 😁  as an alternative to authors and agreed when planning and referring to a release milestone names work much better for a memory-retention perspective.

Reply:
As per poll set up by @barbaricyawps , naming convention was to use bridge names.

Reply:
Yes, agreed. Let's make sure we're more specific here in the RFC to reduce confusion. My suggested wording:

As discussed in the January 26, 2022 general US meeting where a quorum of project steering committee members were present, we will adopt internal codenames to refer to our template and tool releases. We ran an internal poll to determine what the naming convention should be and we agreed they would be named after notable bridges throughout the world. Our codenames are alphabetical, which means that the first release will feature a bridge name that starts with the letter A. For example: Accademia, Ada, Akashi, Akrobaten, Alamillio, Anzac. The following release will have a bridge name that starts with the letter B and so forth. Bridge names will be determined by community vote each release.

Reply:
I made a major mistake and accidentally merged this RFC in prematurely. I'm opening up a new one. I've added this suggested paragraph to a new section of the RFC.